pkg/plugins/backendplugin/manager/manager.go

```diff
@@ -86,6 +88,13 @@ func (m *manager) Register(pluginID string, factory backendplugin.PluginFactoryF
 		}
 	}
 
+	if m.Cfg.AWSAssumeRoleEnabled {
+		hostEnv = append(hostEnv, awsds.ENV_VAR_AssumeRoleEnabled+"=true")
+	}
+	if len(m.Cfg.AWSAllowedAuthProviders) > 0 {
+		hostEnv = append(hostEnv, awsds.ENV_VAR_AllowedAuthProviders+"="+strings.Join(m.Cfg.AWSAllowedAuthProviders, ","))
+	}
```

Comment:
It's confusing that these AWS specifics show up here in otherwise generic plugin management code, could you stick them in an AWS specific function (called from here)? It would make the code easier to follow at least for me.

Reply:
Sounds good, will fix!